### PR TITLE
ut: add mcs validation ut

### PR DESCRIPTION
**Pull request by whitewindmills**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
add mcs validation ut

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Thanks
/assign

**Comment by XiShanYongYe-Chang** [comment]
I would like to invite the owner of the code being tested to take a look. 
Hi @yike21, can you help take a review?
/cc @yike21

File: pkg/webhook/multiclusterservice/validating_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import (
+	"strings"
+	"testing"
+
+	networkingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/networking/v1alpha1"
+)
+
+func TestValidateMultiClusterServiceSpec(t *testing.T) {
+	validator := &ValidatingAdmission{}
+
```
Comment (yike21) [inline pkg/webhook/multiclusterservice/validating_test.go:12]:
```suggestion
specFld := field.NewPath("spec")
```

File: pkg/webhook/multiclusterservice/validating_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	tests := []struct {
+		name        string
+		mcs         *networkingv1alpha1.MultiClusterService
+		expectedErr string
```
Comment (yike21) [inline pkg/webhook/multiclusterservice/validating_test.go:16]:
```suggestion
		expectedErr field.ErrorList
```

File: pkg/webhook/multiclusterservice/validating_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+					Types: []networkingv1alpha1.ExposureType{
+						networkingv1alpha1.ExposureTypeLoadBalancer,
+						networkingv1alpha1.ExposureTypeCrossCluster,
+					},
+					Range: networkingv1alpha1.ExposureRange{
+						ClusterNames: []string{"member1", "member2"},
+					},
+				},
+			},
+			expectedErr: "",
```
Comment (yike21) [inline pkg/webhook/multiclusterservice/validating_test.go:41]:
```suggestion
			expectedErr: field.ErrorList{},
```

File: pkg/webhook/multiclusterservice/validating_test.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+					Types: []networkingv1alpha1.ExposureType{
+						networkingv1alpha1.ExposureTypeLoadBalancer,
+						networkingv1alpha1.ExposureTypeLoadBalancer,
+					},
+					Range: networkingv1alpha1.ExposureRange{
+						ClusterNames: []string{"member1"},
+					},
+				},
+			},
+			expectedErr: `Duplicate value: "foo"`,
```
Comment (yike21) [inline pkg/webhook/multiclusterservice/validating_test.go:66]:
```suggestion
			expectedErr: field.ErrorList{field.Duplicate(specFld.Child("ports").Index(1).Child("name"), "foo")},
```

File: pkg/webhook/multiclusterservice/validating_test.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+					},
+					Types: []networkingv1alpha1.ExposureType{
+						networkingv1alpha1.ExposureTypeLoadBalancer,
+					},
+					Range: networkingv1alpha1.ExposureRange{
+						ClusterNames: []string{"member1"},
+					},
+				},
+			},
+			expectedErr: "must be between 1 and 65535, inclusive",
```
Comment (yike21) [inline pkg/webhook/multiclusterservice/validating_test.go:86]:
```suggestion
			expectedErr: field.ErrorList{field.Invalid(specFld.Child("ports").Index(0).Child("port"), int32(163121), validation.InclusiveRangeError(1, 65535))},
```

File: pkg/webhook/multiclusterservice/validating_test.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+					},
+					Types: []networkingv1alpha1.ExposureType{
+						"",
+					},
+					Range: networkingv1alpha1.ExposureRange{
+						ClusterNames: []string{"member1"},
+					},
+				},
+			},
+			expectedErr: "ExposureType Error",
```
Comment (yike21) [inline pkg/webhook/multiclusterservice/validating_test.go:106]:
```suggestion
			expectedErr: field.ErrorList{field.Invalid(specFld.Child("types").Index(0), networkingv1alpha1.ExposureType(""), "ExposureType Error")},
```

File: pkg/webhook/multiclusterservice/validating_test.go [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+					},
+					Types: []networkingv1alpha1.ExposureType{
+						networkingv1alpha1.ExposureTypeCrossCluster,
+					},
+					Range: networkingv1alpha1.ExposureRange{
+						ClusterNames: []string{"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"},
+					},
+				},
+			},
+			expectedErr: "must be no more than 48 characters",
```
Comment (yike21) [inline pkg/webhook/multiclusterservice/validating_test.go:126]:
```suggestion
			expectedErr: field.ErrorList{field.Invalid(specFld.Child("range").Child("clusterNames").Index(0), "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "must be no more than 48 characters")},
```

File: pkg/webhook/multiclusterservice/validating_test.go [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+				} else if !strings.Contains(errStr, tt.expectedErr) {
+					t.Errorf("expected to contain:\n  %s\ngot:\n  %s", tt.expectedErr, errStr)
+				}
+			} else {
+				if tt.expectedErr != "" {
+					t.Errorf("unexpected no error, expected to contain:\n  %s", tt.expectedErr)
+				}
+			}
+		})
+	}
```
Comment (yike21) [inline pkg/webhook/multiclusterservice/validating_test.go:146]:
```suggestion
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := validator.validateMultiClusterServiceSpec(tt.mcs); !reflect.DeepEqual(got, tt.expectedErr) {
				t.Errorf("validateMultiClusterServiceSpec() = %v, want %v", got, tt.expectedErr)
			}
		})
	}
```

**Review by yike21 — COMMENTED** [review]
Hi, @whitewindmills Thanks for your work! I find a similar ut `pkg/webhook/federatedresourcequota/validating_test.go` and I made some modifications based on it to make the ut clearer.

**Comment by whitewindmills** [comment]
thanks @yike21 :relaxed:

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks
LGTM
/cc @RainbowMango

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**